process/helpers/scAddressCreator.go

```diff
@@ -0,0 +1,49 @@
+package helpers
+
+import (
+	"encoding/binary"
+
+	"github.com/ElrondNetwork/elrond-go/hashing/keccak"
+)
+
+var numInitCharsForScAddr = 10
+var shardIdLength = 2
+var vmTypeLen = 2
+var vmType = []byte{5, 0} // WASM
+var hasher = keccak.Keccak{}
+
+// CreateScAddress will return a new smart contract address for the given address and its nonce
+func CreateScAddress(owner []byte, nonce uint64) ([]byte, error) {
```

Comment:
I would have instantiated a blockchain hook implementation for this.

Reply:
No need for this. As discussed, this function from the blockchainHook is not coupled with the rest of the component so we should extract it to a new component. If/when we'll have this external component, I will use it here.